src/service-worker/core.js

```diff
@@ -241,40 +363,100 @@ const cachePromise = self.caches.open('cdn-js').then(result => {
  * @param {!Cache} cache
  * @param {!Request} request
  * @param {string} requestPath the pathname of the request
- * @param {RtvVersion} requestVersion the version of the request
+ * @param {!RtvVersion} requestVersion the version of the request
  * @return {!Promise<!Response>}
  * @visibleForTesting
  */
-export function fetchAndCache(cache, request, requestPath, requestVersion) {
+export function fetchJsFile(cache, request, requestPath, requestVersion) {
   // TODO(jridgewell): we should also fetch this requestVersion for all files
   // we know about.
-  return fetch(request).then(response => {
-    // Did we receive a valid response (200 <= status < 300)?
-    if (response && response.ok) {
-      // You must clone to prevent double reading the body.
-      cache.put(request, response.clone());
-
-      // Prune old versions of this file from the cache.
-      // This intentionally does not block the request resolution to speed
-      // things up. This is likely fine since you don't have multiple
-      // `<script>`s with the same `src` on a page.
-      cache.keys().then(requests => {
-        for (let i = 0; i < requests.length; i++) {
-          const request = requests[i];
-          const url = request.url;
-          if (requestPath !== pathname(url)) {
-            continue;
-          }
-          if (requestVersion === rtvVersion(url)) {
-            continue;
-          }
+  return fetchAndCache(cache, request).then(response => {
+    // Fetch all diversions of this file.
+    // This intentionally does not block the request resolution to speed
+    // things up.
+    diversions(cache).then(diversions => {
```

Comment:
I thought about that, and thought maybe (slightly) [delaying the initial requests](https://github.com/ampproject/amphtml/pull/7914/files#diff-bbb754c6f5a7768c4e0956b433d2cc4fR485) to batch them together would be a better approach. It would help this case, and would allow us to be smarter about which cached version to serve.

But, this isn't in the critical path. Notice this line isn't returned, the line two under this statement is.